client/app/components/auth/sign-in.js

```diff
@@ -0,0 +1,35 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+
+const noop = async () => {};
+
+export default class AuthSignInComponent extends Component {
+  @service
+  router;
+
+  email;
+
+  searchContacts = this.args.searchContacts || noop;
+
+  @action
+  async startLogin(loginEmail) {
+    const {
+      isNycidValidated,
+      isNycidEmailRegistered,
+      isCityEmployee,
+    } = await this.args.searchContacts(loginEmail);
+
+    if (isNycidEmailRegistered) {
+      // we can only know email validation after they logged in once.
+      // if it's null, it means we don't know because they've never logged in before (null).
+      if (isNycidValidated === true || isNycidValidated === null) {
+        this.router.transitionTo('auth.login', { queryParams: { loginEmail, isCityEmployee } });
+      } else if (isNycidValidated === false) {
```

Comment:
Is it necessary to check `isNycidValidated` here? Since we can't know if an email is validated until after the first successful login, perhaps `startLogin()` doesn't even need to check for it. We could instead check if the email is validated upon login, and if it's not validated `doLogout()` and redirect the user to the `auth.validate` route.

Reply:
Yeah! I think that makes sense.